### PR TITLE
Implement boost param

### PR DESCRIPTION
**Pull request by nikitagromov**
Pivotal story id [#139740111](https://www.pivotaltracker.com/story/show/139740111)

### CODE REVIEW

File: nefertari/es_query.py [hunk 1]
```diff
@@ -8,6 +9,508 @@ def pop(self, index=None):
...
+            raise ValueError('Wrong numbers of parentheses. Query string could not be parsed')
+
+        return head.parse()
+
+
+class Tokenizer:
+    query_keywords = {'AND', 'OR', 'NOT'}
+
+    def __init__(self):
+        pass
```
Comment (nmenardg) [inline nefertari/es_query.py]: Is this necessary?

File: nefertari/es_query.py [hunk 2]
```diff
@@ -8,6 +9,508 @@ def pop(self, index=None):
...
+
+    @staticmethod
+    def append(buffer, tokens):
```
Comment (nmenardg) [inline nefertari/es_query.py]: Why not use `@classmethod`?

File: nefertari/es_query.py [hunk 3]
```diff
@@ -8,6 +9,508 @@ def pop(self, index=None):
...
+
+        value = buffer.get_value()
+
+        if value not in Tokenizer.query_keywords:
```
Comment (nmenardg) [inline nefertari/es_query.py]: ...and then `cls.query_keywords`

File: nefertari/es_query.py [hunk 4]
```diff
@@ -8,6 +9,508 @@ def pop(self, index=None):
...
+            if value == 'NOT':
+                last_value = tokens.pop()
+                tokens.append(' '.join([last_value, value]))
+            else:
+                tokens.append(value)
+            buffer.clean(with_cache=True)
+            buffer.reset_spaces()
+
+    @staticmethod
+    def tokenize(query_string):
```
Comment (nmenardg) [inline nefertari/es_query.py]: Same here

File: nefertari/es_query.py [hunk 5]
```diff
@@ -8,6 +9,508 @@ def pop(self, index=None):
...
+
+        if buffer:
+            Tokenizer.append(buffer, tokens)
+
+        Tokenizer._remove_needless_parentheses(tokens)
+
+        return tokens
+
+    @staticmethod
+    def _remove_needless_parentheses(tokens):
```
Comment (nmenardg) [inline nefertari/es_query.py]: Same here

File: nefertari/es_query.py [hunk 6]
```diff
@@ -8,6 +9,508 @@ def pop(self, index=None):
...
+        self.spaces_counter += 1
+
+    def get_spaces(self):
+        return self.spaces_counter * ' '
+
+    def get_value(self):
+        return self.value
+
+    def get_cache(self):
+        return self.cached
```
Comment (nmenardg) [inline nefertari/es_query.py]: Why not use `self.value` and `self.cached` directly?

File: nefertari/es_query.py [hunk 7]
```diff
@@ -8,6 +9,508 @@ def pop(self, index=None):
...
+        self.boost_params = boost_params
+
+    def rule(self, term):
+        return term.field in self.boost_params or (term.field == '_all' and self.boost_params)
+
+    @staticmethod
+    def next():
+        return FinalProcessor
+
+    def apply(self, term):
```
Comment (nmenardg) [inline nefertari/es_query.py]: I think you could define those methods in the BaseProcessor as `@abstractmethod`

**Review by nmenardg — CHANGES_REQUESTED** [review]

File: nefertari/es_query.py [hunk 8]
```diff
@@ -8,6 +9,508 @@ def pop(self, index=None):
...
+        def find_conflicts(value):
+            counter = 0
+            for item in graph.values():
+                if item == value:
+                    counter += 1
+
+                if counter > 1:
+                    return True
+            else:
+                return False
```
Comment (nmenardg) [inline nefertari/es_query.py]: As Codacy highlighted, the `return False` could be directly in the function, after the `for`

**Review by nmenardg — APPROVED** [review]

File: tests/test_es_query.py [hunk 9]
```diff
@@ -460,10 +468,34 @@ def test_query_with_all_and_boost(self):
...
+            assert item in expected_terms
+            expected_terms.remove(item)
+
+        # {'bool': {'must': [{'bool': {'must': [{'bool': {
+        #     'should': [{'wildcard': {'_all': '*name*'}},
+        #                {'wildcard': {'name': {'value': '*name*', 'boost': '10'}}},
+        #                {'wildcard': {'description': {'value': '*name*', 'boost': '5'}}}],
+        #     'minimum_should_match': 1
+        # }}, {
+        #     'term': {'obj_status': 'active'}}]}}]}}
```
Comment (nmenardg) [inline tests/test_es_query.py:503]: I think you forgot commented code here

**Review by nmenardg — CHANGES_REQUESTED** [review]

File: tests/test_es_query.py (hunk 9 above)
Comment (nikitagromov) [reply tests/test_es_query.py:503]: I left this here for readability

Comment (nmenardg) [reply tests/test_es_query.py:503]: Please add a small explanation text

**Review by nmenardg — APPROVED** [review]

**Pull request merged**